src/utils.ts

```diff
@@ -109,6 +109,26 @@ export function joinURL (base: string, ...input: string[]): string {
   return url
 }
 
+export function withHttp (input: string): string {
+  return withProtocol(input, 'http:')
+}
+
+export function withHttps (input: string): string {
+  return withProtocol(input, 'https:')
+}
+
+export function withoutProtocol (input: string): string {
+  const parsed = parseURL(input)
+  parsed.protocol = ''
+  return stringifyParsedURL(parsed, true)
+}
```

Comment:
We could use a regex to match protocol part only. Early returning when protocol not exists and simply remove protocol with regex match position without expensive parse/stringify which can also potentially change formatting in edge cases